src/bin/pgcopydb/table-data.c

```diff
@@ -365,6 +365,15 @@ copydb_copy_supervisor(CopyDataSpec *specs)
 		return false;
 	}
 
+	/*
+	 * If we're resuming, try creating indexes that are not processed.
+	 */
+	if (specs->resume && !copydb_create_remaining_indexes(specs))
+	{
+		/* errors have already been logged */
+		return false;
+	}
+
```

Comment:
How does that work with tables that have not been taken care of yet?

Reply:
@dimitri copydb_create_remaining_indexes would be called after completion of table copy workers.,when the control reaches here, all the tables should have been copied already.

Reply:
I see. I expected that processing to happen on table-by-table basis, in order to benefit from the usual amount of concurrency between copy workers and index workers. Does it make sense to check for remaining indexes for each table even when skipping the table on the basis that it's been already done? We might still need that piece of code here to catch indexes remaining to be created in case that all the table were done in the previous round I suppose?

Reply:
@dimitri You are right. We need this piece of code to handle if all the tables were done already. I'm also wondering whether we can resume the respective workers with aborted commands. Basically, populating the queue with the entries from process & summary.

Reply:
@dimitri What if we let the control proceed through all the tables and then through all the indexes in the original execution order, regardless of their completion status, skipping creation if it has already been completed? 

The best analogy for this is how make processes its targets. It reviews all the targets and generates the output if it doesn't exist, regardless of the current existence of the target output.

Reply:
@dimitri Pushed a new commit which allows table-data worker to continue signalling index workers even if the table data copy is done. Which would essentially retain the original flow with same concurrency pattern.